flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetInputFormat.java

```diff
@@ -160,13 +168,16 @@ public void setFilterPredicate(FilterPredicate filterPredicate) {
     public void open(FileInputSplit split) throws IOException {
         // reset the flag when open a new split
         this.skipThisSplit = false;
+        // TODO why is this needed to read the parquet schema from file in open() ? Can't we use the
+        //  already extracted schema (extracted in the constructor) ?
```

Comment:
See above.

Reply:
Yes, clearer now, thanks. But still I think there is a bug, see my comment at the end of the review.